### PR TITLE
refactor "Installing Kubeflow" page

### PR DESCRIPTION
**Pull request by thesuperzapper**
Closes https://github.com/kubeflow/website/issues/2590

As discussed in Issue https://github.com/kubeflow/website/issues/2590, this is a rewrite for the "Installing Kubeflow" page.

### CODE REVIEW

**Comment by thesuperzapper** [comment]
@RFMVasconcelos @cvenets

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:43]: I would suggest removing links that drive users outside of docs to things like the main vendor websites to keep the UX clean.

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:79]: <td>AKS, EKS, GKE and more K8s</a></td>

or 

        <td>Kubernetes agnostic</a></td>

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:77]: Maybe `Kubeflow Charmed Operators`

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:67]: Here it should be `MicroK8s Kubeflow add-on`

File: content/en/docs/started/installing-kubeflow.md
Comment (joeliedtke) [inline content/en/docs/started/installing-kubeflow.md:42]: Define any abbreviations that you use, such as AWS, EKS, or GKE. For example, the first time you use GKE in a document, refer to it as:

Google Kubernetes Engine (GKE)

File: content/en/docs/started/installing-kubeflow.md
Comment (joeliedtke) [inline content/en/docs/started/installing-kubeflow.md:19]: Order the paths in the best order for a new user. Would you recommend that a new Kubeflow user create their own distribution, or would you recommend that they start with one of the existing distributions? (I would personally recommend that they start with an existing distribution.)

Comment (joeliedtke) [inline content/en/docs/started/installing-kubeflow.md:19]: I would recommend removing the  "Path #:" prefix from the headings. Also, do not use slang such as "roll your own" to describe creating a distribution. Colloquialisms and slang can make documentation more difficult for non-english speaking readers to understand. Here are some suggestions for the headings:

- Create a custom distribution
- Create your own distribution
- Customize a distribution
- Select a distribution 
- Choose a distribution
- Use a distribution

File: content/en/docs/started/installing-kubeflow.md
Comment (joeliedtke) [inline content/en/docs/started/installing-kubeflow.md:30]: Consider adding a sentence or two here that describes what the user should do.

File: content/en/docs/started/installing-kubeflow.md
Comment (joeliedtke) [inline content/en/docs/started/installing-kubeflow.md:53]: GCP -> Google Cloud

File: content/en/docs/started/installing-kubeflow.md
Comment (joeliedtke) [reply content/en/docs/started/installing-kubeflow.md:43]: I agree that linking to the main vendor website may not be helpful. I think that it could be helpful to keep the link to the Kubernetes environment could be helpful. Consider linking to the docs instead of the product pages.

**Review by joeliedtke — COMMENTED** [review]
Added a few suggestions

**Comment by shannonbradshaw** [comment]
@joeliedtke I think maybe you want to take a look at this PR instead: Most of what you're commenting on, I've already fixed on this PR: https://github.com/kubeflow/website/pull/2625

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: @joeliedtke I agree, and have removed the links to under the "Maintainers" section.

@RFMVasconcelos Regarding linking to external websites, as I discussed in https://github.com/kubeflow/website/issues/2590#issuecomment-820888767, we should aim to not be hosting any distribution specific docs on `kubeflow.org`.

Users should go to the "Installing Kubeflow" page for a list of current distributions, and get links to them.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have replaced this with `Conformant Kubernetes`

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: Fixed

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have made this change

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have made this change

Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have made this change, and replaced the names with:
* `Packaged distributions`
* `Directly use the manifests`

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have made this change

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:30]: @joeliedtke I have added some bullet points for what I would say.

What are your thoughts?

**Comment by thesuperzapper** [comment]
@RFMVasconcelos @joeliedtke I have made changes based off the comments, any further thoughts?

After we agree on the bullet points under each section, we can write the actual text.

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:82]: I don't see a huge issue with pointing to external docs here, but, in any case, there is documentation on `distributions` upstream that needs to be maintained. Hence, pointing to internal docs might be beneficial. This:
1. Raises the bar for "becoming a distribution" leading to quality
2. Motivates distribution owners to keep their docs up to date 

> Why is there a need for `distributions` docs upstream? 

Some distributions have it leaving any distribution that does not at a disadvantage.

**Review by RFMVasconcelos — COMMENTED** [review]
@thesuperzapper thank you for leading this! I have left a few comments throughout :)

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:2]: I think we should put the stuff about `Components` in the `Getting Started/Overview` page (which we will rewrite after we merge this).

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:82]: @RFMVasconcelos I have raised an issue to discuss the removal of the distribution-specific docs from kubeflow.org, which will mean the `Getting Started/Installing Kubeflow` page will be the only place which references distributions.

But I agree that if there is currently a doc page on kubeflow.org, we should link there for now.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have removed all the links to platforms.

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [reply content/en/docs/started/installing-kubeflow.md:2]: The downside I see of this approach is that we have effectively 1 gateway page that users arrive at when they click on `Getting Started` from the home page. 

Having the information about components on a separate page would require the user to find that second page. Having it upon scroll appears to be the natural place for it IMO. But open to more comments on this :)

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [reply content/en/docs/started/installing-kubeflow.md]: @thesuperzapper ok! Happy to discuss this on a separate thread so we can wrap this higher priority item! :)

**Comment by RFMVasconcelos** [comment]
@thesuperzapper Thank you for your last commits, this is shaping up to be a great overview! :tada: 

Please see [this comment](https://github.com/kubeflow/website/pull/2611#discussion_r617326511). Aside from that, I think we are ready to write the missing descriptions and effectively merge this. WDYT?

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:19]: Proposal for overview text (edited from #2625):

> Kubeflow is an MLOps platform that aggregates a set of [core components](link) with [external add-ons](link) to orchestrate Machine Learning processes on Kubernetes.

> Kubeflow supports ML workflows from data preparation through training and deployment. It enables a scalable pipeline structure that helps ensure containerized steps with defined dependencies. Kubeflow extends Kubernetes’ ability to run independent and configurable systems with a curated set of compatible tools and frameworks specific for ML. 

Proposal for 2 paths text (edited from #2625):

> There are two options for getting up and running with Kubeflow. You can select one of the supported Kubeflow distributions from the table below, or you can roll your own version, directly from [manifests](https://github.com/kubeflow/manifests).

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:30]: Proposal (edited from #2625):

> Kubeflow distributions are developed and supported by their respective maintainers. The community does not currently endorse or certify any of the distributions, fostering openness and healthy competition. See the table below for a list of options and links to documentation.

@thesuperzapper wdyt? maybe too much? :)

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [inline content/en/docs/started/installing-kubeflow.md:118]: Proposal (edited from #2625):

> The Kubeflow Manifests Working Group maintains up-to-date manifests for each official Kubeflow component and enables you to install all components or customize an installation to include just the components you need. To install Kubeflow on your own, see the Kubeflow Manifests [installation instructions](https://github.com/kubeflow/manifests#installation).

**Review by RFMVasconcelos — COMMENTED** [review]
@shannonbradshaw I think we should update this page centrally here. I have adapted pieces from your #2625 PR to complete this one so we can maximize output and take advantage of everyone's willingness to contribute.

@thesuperzapper can you review the text proposals and, with your edits, commit so we can get to a final PR?

**Comment by shannonbradshaw** [comment]
> @shannonbradshaw I think we should update this page centrally here. I have adapted pieces from your #2625 PR to complete this one so we can maximize output and take advantage of everyone's willingness to contribute.
> 

@RFMVasconcelos sounds good! Thanks for all the work, @thesuperzapper and @RFMVasconcelos.

File: content/en/docs/started/installing-kubeflow.md
Comment (zijianjoy) [inline content/en/docs/started/installing-kubeflow.md:106]: Curious if the community is interested in adding an `on-prem kubeflow installation` guide?

File: content/en/docs/started/installing-kubeflow.md
Comment (zijianjoy) [inline content/en/docs/started/installing-kubeflow.md:120]: Should we direct user to individual component repo for those roll-your-own installation? I would expect GitHub issues coming in to `kubeflow/manifests` where might not get enough attention by individual components' WGs.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:2]: Lets just leave it is for now, and revisit it as we rewrite the overview page.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:29]: I have included something very similar to your suggestion

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:105]: Lets just leave this a distributions for now, but if there is an on-prem distribution, we could link it here.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I agree that we need a good pathway to help users direct component-specific issues to the right place, but I dont think the install guide is the right place for that.

Especially given most users should not be installing directly from the manifests, and should be using a supported distribution.

**Comment by thesuperzapper** [comment]
@RFMVasconcelos I have incorporated your suggestions, and I think this is ready to `/approve`.

Obviously we will iterate on this page, but we should try and get this merged very soon, so it can be linked to in the 1.3 blogs.

**Comment by shannonbradshaw** [comment]
@thesuperzapper Arrikto Enterprise Kubeflow should be listed among the distributions. I see it's not currently in your PR. Here's the content for a page to include under distributions.

+++
title = "Arrikto Enterprise Kubeflow"
description = "Kubeflow distribution with additional automation, reproducibility, portability, and security features"
weight = 50
+++

The <a href="https://www.arrikto.com/enterprise-kubeflow/" target="_blank">Arrikto Enterprise Kubeflow (EKF)</a> distribution extends the capabilities of the Kubeflow platform with additional automation, reproducibility, portability, and security features.
- *Automation*: Orchestrate your end-to-end ML workflow with a click of a button. Start by tagging cells in Jupyter Notebooks to define pipeline steps, hyperparameter tuning, GPU usage, and metrics tracking. Click a button to create pipeline components and KFP DSL, resolve dependencies, inject data objects into each step, deploy the data science pipeline, and serve the best model. Or use the Kale SDK to do all the above with your preferred IDE.
- *Reproducibility*: Snapshot pipeline code, libraries, and data for every step with Arrikto’s Rok data management platform. Roll back to any machine learning pipeline step at it’s exact execution state for easy debugging. Collaborate with other data scientists through a Git-style publish/subscribe versioning workflow.
- *Portability*: Deploy and upgrade your Kubeflow environment with a proven GitOps process across all major public clouds, and on-prem infrastructure. Move ML workflows seamlessly across with Rok Registry.
- *Security*: Manage teams and user access via GitLab or any ID provider via Istio/OIDC. Isolate user ML data access within their own namespace while enabling notebook and pipeline collaboration in shared namespaces. Manage secrets and credentials securely, and efficiently.

**Comment by shannonbradshaw** [comment]
@thesuperzapper the column values for the table entry for the EKF distribution should be:

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:39]: Amazon Web Services (AWS)

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:46]: Microsoft Azure

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:53]: Google Cloud Platform (GCP)

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:68]: OpenShift

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:67]: Red Hat

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:66]: OpenShift

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:89]: MicroK8s

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:94]: Arrikto MiniKF

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:96]: AWS Marketplace, GCP Marketplace, Vagrant

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:99]: `<tr>
        <td><a href="/docs/distributions/ekf/">Arrikto Enterprise Kubeflow</a></td>
        <td>Arrikto</td>
        <td>EKS, 
            AKS,
            GKE 
        </td>
      </tr>`

**Review by shannonbradshaw — CHANGES_REQUESTED** [review]
Requesting a few changes.

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [reply content/en/docs/started/installing-kubeflow.md:39]: Best practice is to define acronyms, even those that are very well known.

File: content/en/docs/started/installing-kubeflow.md
Comment (shannonbradshaw) [inline content/en/docs/started/installing-kubeflow.md:108]: This method is for advanced users. The Kubeflow community will not support environment-specific issues. If you need support, please consider using a [packaged distribution](#packaged-distributions) of Kubeflow.

Comment (shannonbradshaw) [reply content/en/docs/started/installing-kubeflow.md:108]: Fixed the comma splice.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:53]: The google guys asked to make it Google Cloud (as that is the "entity" which is maintaining the distribution), @Bobgy thoughts?

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:67]: Currently, this distribution is maintained by IBM Cloud, (not RedHat) @animeshsingh can verify.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have implemented something similar to this.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have implemented something similar to this.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:99]: Lets keep this PR to refactoring what was already there, if you want to add another doc page, you can just raise a PR which updates the "Installing Kubeflow" page, after this has been merged.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:39]: I am treating the first column, as the actual "name" of the distribution, which is precisely "Kubeflow on AWS".

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:94]: I am fine to change it, but is that the "name" of the distribution?

Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:94]: @shannonbradshaw I am fine to change it, but is that the "name" of the distribution?

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: done

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: done

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: done

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: done

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:39]: I now realise you were referring to the "maintainers" column, I have changed this to "Amazon Web Services".

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md:46]: done

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: done

**Comment by thesuperzapper** [comment]
@shannonbradshaw you mentioned wanting to add the "Arrikto Enterprise Kubeflow" distribution, please raise a separate PR (after this one is merged), as this PR is only for refactoring the "Installing Kubeflow" page, not adding a new distribution doc.

File: content/en/docs/started/installing-kubeflow.md
Comment (thesuperzapper) [reply content/en/docs/started/installing-kubeflow.md]: I have made this change, feel free to change it later.

**Comment by thesuperzapper** [comment]
@RFMVasconcelos  I think this is good enough to merge now so we have a proper "Installing Kubeflow" page up by the time 1.3 is dropped tonight. 

We can keep iterating on it over the next while if needed.

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [reply content/en/docs/started/installing-kubeflow.md:100]: I think there are multiple on-prem options, so we could at most tag them. Providing 1 guide will likely favour one option.

File: content/en/docs/started/installing-kubeflow.md
Comment (RFMVasconcelos) [reply content/en/docs/started/installing-kubeflow.md:2]: Ok! LGTM given the urgency

**Review by RFMVasconcelos — COMMENTED** [review]
@thesuperzapper @shannonbradshaw for the interest of time I agree with @thesuperzapper that we should merge this and then keep on iterating in further PRs. 

For reference, this will merge to the `master` branch. For these changes to be effective we need @Bobgy or @joeliedtke to change the config in netlify so that the website points to that branch.

/lgtm 
/approve

**Comment by Bobgy** [comment]
Adding some late comments:
1. I think we are missing a redirect rule from the old page to the new page
2. https://deploy-preview-2611--competent-brattain-de2d6d.netlify.app/docs/started/getting-started/ is broken, we should update the "getting started" button url
3. the link to manifest README should better pin to 1.3.0 release once we pin it

File: content/en/docs/started/installing-kubeflow.md
Comment (Bobgy) [reply content/en/docs/started/installing-kubeflow.md:53]: Yes, that's right. Now we stick to calling it Google Cloud.

**Comment by RFMVasconcelos** [comment]
Sorry if I prematurely merged this one @Bobgy, but wanted to accelerate the progress, allowing iteration in further PRs. :)

Blockers: 
1. I noticed and fixed this in #2635, was a BIG problem :joy: 
2. Also fixed in #2635 (I think, please verify :) )
3. +1
4. +1 - too many columns may wrap but I think this would be useful! We can find a sensible way to make it fit.

File: content/en/docs/started/installing-kubeflow.md
Comment (zijianjoy) [reply content/en/docs/started/installing-kubeflow.md:100]: I agree that we can discuss later on. We don't need to take action on linking them for this release.

File: content/en/docs/started/installing-kubeflow.md
Comment (zijianjoy) [reply content/en/docs/started/installing-kubeflow.md]: Got it, thank you @thesuperzapper for the explanation!

**Pull request merged**
